core-common/src/test/java/org/zalando/nakadi/annotations/validation/DataLakeAnnotationsTest.java

```diff
@@ -31,14 +32,28 @@ public void prepareValidator() {
         validator = Validation.buildDefaultValidatorFactory().getValidator();
     }
 
+    @Test
+    public void whenMaterializationEventFormatIsWrongThenFail() {
+        final var annotations = Map.of(
+                DataLakeAnnotationValidator.MATERIALISE_EVENTS_ANNOTATION, "1 day"
+        );
+        final Set<ConstraintViolation<TestClass>> result = validator.validate(new TestClass(annotations));
+        assertTrue(result.stream().anyMatch(r -> r.getMessage().equals("Field " +
+                DataLakeAnnotationValidator.MATERIALISE_EVENTS_ANNOTATION
+                + " is not valid. Provided value: \""
+                + annotations.get(DataLakeAnnotationValidator.MATERIALISE_EVENTS_ANNOTATION)
+                + "\". Possible values are: \"on\" or \"off\".")));
```

Comment:
Imo, matching the error messages exactly _as is_ is not maintainable long term.
You can keep the test more flexible, by checking that the error message contains some specific substring. In this case, for example, check that the message mentions the annotation's name.

Reply:
Fixed, please see the comment below.